superset-frontend/spec/javascripts/dashboard/util/isValidChild_spec.ts

```diff
@@ -136,20 +136,22 @@ describe('isValidChild', () => {
     invalidExamples.forEach((example, exampleIdx) => {
       let childDepth = 0;
       example.forEach((childType, i) => {
-        const shouldTestChild = Array.isArray(childType);
-
-        if (i > 0 && shouldTestChild) {
+        // should test child
+        if (i > 0 && Array.isArray(childType)) {
```

Comment:
Currently type narrowing using `Array.isArray` is limited to when the it is inside an `if` condition